api/v1beta1/types.go

```diff
@@ -84,11 +84,32 @@ type FilterByNeutronTags struct {
 	NotTagsAny []NeutronTag `json:"notTagsAny,omitempty"`
 }
 
+// SecurityGroupFilter describes the data needed to identify an SG to use. If ID is provided it is required that all other fields are unset.
+// +kubebuilder:validation:XValidation:rule="has(self.id) ? !has(self.name) : true",message="when ID is set you cannot set other options"
+// +kubebuilder:validation:XValidation:rule="has(self.id) ? !has(self.description) : true",message="when ID is set you cannot set other options"
+// +kubebuilder:validation:XValidation:rule="has(self.id) ? !has(self.projectID) : true",message="when ID is set you cannot set other options"
 type SecurityGroupFilter struct {
-	ID          string `json:"id,omitempty"`
-	Name        string `json:"name,omitempty"`
-	Description string `json:"description,omitempty"`
-	ProjectID   string `json:"projectID,omitempty"`
+	// The ID of the desired security group. If ID is provided, the other filters cannot be provided.
```

Comment:
should this rule also be applies to tags?

Reply:
Same here, I couldn't add tags without exceeding CEL rule budget.